validphys2/src/validphys/utils.py

```diff
@@ -5,14 +5,52 @@
 @author: Zahari Kassabov
 """
 import contextlib
+import functools
 import pathlib
 import shutil
 import tempfile
+from typing import Any, Sequence, Mapping
 
 import numpy as np
 from validobj import ValidationError, parse_input
 
 
+# Since typing.Hashable doesn't check recursively you actually
+# have to try hashing it.
+def is_hashable(obj):
+    try:
+        hash(obj)
+        return True
+    except:
+        return False
+
+
+def immute(obj: Any):
+    # So that we don't infinitely recurse since frozenset and tuples
+    # are Sequences.
+    if is_hashable(obj):
+        return obj
+    elif isinstance(obj, Mapping):
+        return frozenset([(immute(k), immute(v)) for k, v in obj.items()])
+    elif isinstance(obj, Sequence):
+        return frozenset([immute(i) for i in obj])
```

Comment:
```suggestion
        return tuple(immute(i) for i in obj)
```

It looks like a bad idea to destroy the order using a frozenset, which doesn't keep order. Functions  dealing with a sequence typically assign meaning to it. I think you may want a tuple here instead.

Reply:
You're right. I still suffer from breaking tests though, I may have to reintroduce the frozendict anyway...

Reply:
FYI, I am looking into this now!